app/features/signup/singup-form.tsx

```diff
@@ -0,0 +1,131 @@
+import { useOpenSecret } from '@opensecret/react';
+import { Link } from '@remix-run/react';
+import { useRef } from 'react';
+import { type SubmitHandler, useForm } from 'react-hook-form';
+import { Button } from '~/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from '~/components/ui/card';
+import { Input } from '~/components/ui/input';
+import { Label } from '~/components/ui/label';
+import { useToast } from '~/hooks/use-toast';
+import { buildEmailValidator } from '~/lib/validation';
+
+type FormValues = { email: string; password: string; confirmPassword: string };
+
+const validateEmail = buildEmailValidator('Invalid email');
+
+export function SignupForm() {
+  const { signUp } = useOpenSecret();
+  const { toast } = useToast();
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors, isSubmitting },
+  } = useForm<FormValues>();
+  const password = watch('password');
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    try {
+      await signUp(data.email, data.password, '');
```

Comment:
This will throw an error if the user is already created - "Email already registered"

Reply:
and we will catch it and show a toast message. what else would you do there?

Reply:
maybe we can change the message to throw out contact the support part but I am not sure we even want to tell them that email is taken (we could because OS returns 409 response for that case) because that would be revealing the emails that are already taken or should we do it anyway?

Reply:
Yea I was just thinking there should be a better error message, otherwise the user just thinks it's broken. I don't know how important that is to keep secret for us specifically, but a lot of apps say "this account is already taken"

Reply:
ok. will do the same

Reply:
done